### ISSUE TITLE
Fix/enhance syntax highlighting for lower case SQL statements

### ISSUE THREAD

**Issue by redtopia** [jcberquist/sublimetext-cfml#119]
Currently, syntax highlighting for lower case sql statements appears to work, but isn't triggered unless the first command is uppercase.

Here's an example:
![sql-syntax](https://user-images.githubusercontent.com/812777/43015270-67ae937e-8c0c-11e8-889f-affe609b97aa.gif)

**Comment by jcberquist** [comment]
The reason this is happening is that the syntax highlighting looks for SQL keywords at the beginning of _any_ string, and if one is found, the string is highlighted as SQL. This allows for highlighting SQL in strings without the complications of trying to determine whether a string is expected to be SQL (e.g. inside of the queryExecute() call). Because of the way this works (on any string) I am a little unsure about using lowercase keywords to detect SQL. An uppercase `SELECT` starting a string makes it very likely that the string is SQL, but I think looking for lowercase keywords makes false positives much more likely.

**Comment by redtopia** [comment]
That makes sense. I wonder if anything could be done to improve the logic. I love the sql syntax highlighting, but all too often I'm working with lowercase strings.

**Comment by redtopia** [comment]
Cool... thanks! I'll give it a try. LMK.

**Comment by jcberquist** [comment]
@redtopia I have added some more lowercase sql highlighting. You can look at [syntax_test_sql_strings.cfm](https://github.com/jcberquist/sublimetext-cfml/blob/master/syntaxes/tests/syntax_test_sql_strings.cfm) to get an idea of what is supported now. Some of what is supported will only be available on multiline strings, but I did try to get some single line support in there as well.

**Comment by redtopia** [comment]
Thanks John. The changes aren't really doing much for my codebase... I have a lot of this going on, and from what I can see, none of our lower case queries are highlighted with your current changes:

		var q = queryExecute(sql="
			select a.companyType as companyTypeID, b.companytype
			from tblCompaniesType a
			    inner join tblcompanytypes b on b.companytypeid =  a.companyType
			where a.companyID = :pk_param
			", 
			params=[{ name="pk_param", cfsqltype="cf_sql_other", value=this.companyID }],
			options={
				datasource:variables.dsn
			}
		);

**Comment by jcberquist** [comment]
@redtopia: that is helpful, thanks. I am not going to be able to catch all cases - it is guessing after all. But I can add support to the "select" matching for column aliases.

**Comment by jcberquist** [comment]
I have added another commit to support basic column aliasing after the select.

**Comment by redtopia** [comment]
That looks a lot better! :+1:

**Comment by KamasamaK** [comment]
I agree that guessing whether a generic string literal is a SQL statement should be done conservatively. However, given this example, it is clear that any string literal value of the `sql` parameter should be a SQL statement.

**Comment by redtopia** [comment]
@KamasamaK, I agree, though I'm using named arguments here, and it's entirely possible to call queryExecute() without using named attributes, which makes the problem a bit more complicated.

**Comment by KamasamaK** [comment]
@redtopia Sure, but only a bit more complicated since when not using named arguments, the first argument would have the same rules applied. Both of these cases were actually addressed for `queryexecute` previously, but I'm not sure when it was removed.
